Update documentation for choreo insights API

RAVEENSR opened the pull request:
## Purpose
Update documentation for choreo insights API

## Goals
Update documentation for choreo insights API

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top"><a href="#customreportovertime">CustomReportOvertime</a></td>
+<td>
+
+Returns summary of hits/ latency data for the inquired time frame grouped into timeSpans
```
```suggestion
Returns the summary of hits or latency data as specified for the selected time duration, grouped by time spans.
```
- What did you mean by "timeSpans" here and was there a reason to have it as one word and in camel case?
- Does it return either the summary of hits or latency as specified by the user? Is so, we should mention that.
- What is the difference between the "time frame" and "time span" here? Are time spans standard time granularities?

NavinYP replied:
- Time spans are standard time granularities. The relevant parameter is declared in camel case in the schema. Other than that there is no need to have it as a single word here in camel case.
- Yes, it returns either the summary of hits or latency depending on the metric specified by the user. 
- Time frame is the 'from x to y' time period that the user specified in the 'timeFilter' and time spans are standard time granularities as mentioned before.

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top"><a href="#metric">Metric</a>!</td>
+<td>
+
+Metric to retrieve data for
```
```suggestion
The metric for which data needs to be retrieved.
```

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top">[<a href="#groupbyfield">GroupByField</a>]!</td>
+<td>
+
+Column/ Field to group the data by
```
The column/field by which the data should be grouped.

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top">[<a href="#string">String</a>]!</td>
+<td>
+
+Value to filter by on the groupByField
```
This is not clear. Is the value returned for the groupByField further filtered by something else?

NavinYP replied:
`where groupByField == groupByValue`

`groupByValue` is the key value to filter on the relevant 'groupByField'

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top"><a href="#customreporttopovertime">CustomReportTopOvertime</a></td>
+<td>
+
+Returns summary of hits/ latency data for the inquired time frame grouped into timeSpans for the top N values
```
See comment on line 1210.

NavinYP replied:
Responded in the relevant comment thread

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top"><a href="#metric">Metric</a>!</td>
+<td>
+
+Metric to retrieve data for
```
The metric for which data needs to be retrieved.

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top">[<a href="#groupbyfield">GroupByField</a>]!</td>
+<td>
+
+Column/ Field to group the data by
```
The column/field by which data needs to be retrieved.

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top">[<a href="#string">String</a>]!</td>
+<td>
+
+Value to filter by on the groupByField
```
This is not clear. Is the value returned for the groupByField further filtered by something else?

NavinYP replied:
`where groupByField == groupByValue`

`groupByValue` is the key value to filter by on the relevant `groupByField`

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top">[<a href="#string">String</a>]</td>
+<td>
+
+Returns a list of distinct group by values for the data available for a selected groupByField
```
```suggestion
Returns a list of values by which you can group the data available for the selected `groupBy` field
```

The sentence is not clear. Gave the suggestion based on how I understood it.

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top">[<a href="#groupbyfield">GroupByField</a>]!</td>
+<td>
+
+Column/ Field to get the distinct values for
```
```suggestion
The column/field for which the API needs to retrieve distinct values.
```

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top">[<a href="#string">String</a>]!</td>
+<td>
+
+Value to filter by on the groupByField
```
This is not clear. Is the value returned for the groupByField further filtered by something else?

NavinYP replied:
`where groupByField == groupByValue`

`groupByValue` is the key value to filter by on the relevant `groupByField`

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1841,6 +2042,150 @@ The granularity value for which data is retrieved.
 
+### CustomReportGroupBy
+
+Represents usage data overtime for the selected paramters for a certain groupByValue
```
```suggestion
Represents the usage data overtime for the selected parameters of a specific `groupByValue`.
```

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1841,6 +2042,150 @@ The granularity value for which data is retrieved.
+<td valign="top"><a href="#id">ID</a>!</td>
+<td>
+
+Distinct group by value used as an ID
```
```suggestion
The distinct group-by value that is used as an ID.
```

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1841,6 +2042,150 @@ The granularity value for which data is retrieved.
+<td valign="top">[<a href="#customreportusage">CustomReportUsage</a>]</td>
+<td>
+
+A list containing the usage values. Returns an empty array if no data is available.
```
Shall we change the second sentence to "This "returns an empty array if no data is available."

NavinYP replied:
ack

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1841,6 +2042,150 @@ The granularity value for which data is retrieved.
+<td valign="top">[<a href="#customreportusage">CustomReportUsage</a>]</td>
+<td>
+
+A list containing the usage values. Returns an empty array if no data is available.
```
Shall we change the second sentence to "This returns an empty array if no data is available."

NavinYP replied:
ack

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1841,6 +2042,150 @@ The granularity value for which data is retrieved.
+
+### CustomReportTopOvertime
+
+Represents usage data overtime for the selected paramters grouped by the top N groupByValues
```
"parameters" is misspelt.

What do you mean by "top N groupByValues"?

NavinYP replied:
Data is grouped by the values which has the most number of hits in the relevant column or field.

The number of top values is hard coded at the moment as `5`. So this could be changed to 'grouped by the top 5 groupByValues'

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1841,6 +2042,150 @@ The granularity value for which data is retrieved.
+<td valign="top">[<a href="#customreportgroupby">CustomReportGroupBy</a>]</td>
+<td>
+
+A list containing the usage values grouped by the selected groupByField. Returns an empty array if no data is
```
Shall we change the second sentence to "This returns an empty array if no data is available."?

NavinYP replied:
ack

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -4236,6 +4581,184 @@ Target-related errors.
 
+### GroupByField
+
+Represents an groupByField used in CustomReports.
```
Suggestion:

"Represents a `groupByField` parameter used in `CustomReports`."

When you use a term in this format (e.g., groupByField), have it in code format and state what it is (e.g., `groupByField` parameter). Please do the same to `CustomReports` as well as all the similar terms in this document.

NavinYP replied:
ack

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -4236,6 +4581,184 @@ Target-related errors.
+<td valign="top"><strong>HIT_COUNT</strong></td>
+<td>
+
+Number of API calls
```
```suggestion
The number of API calls.
```

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -4236,6 +4581,184 @@ Target-related errors.
+<td valign="top"><strong>API_ERRORS</strong></td>
+<td>
+
+Number of hits with API errors returning
```
```suggestion
The number of hits for which API errors are returned.
```

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -4236,6 +4581,184 @@ Target-related errors.
+<td valign="top"><strong>TARGET_ERRORS</strong></td>
+<td>
+
+Number of hits with target errors returning
```
```suggestion
The number of hits for which target errors are returned.
```

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top"><a href="#string">String</a>!</td>
+<td>
+
+Value of the On-Prem key
```
```suggestion
The value of the On-Prem key.
```

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -1162,6 +1162,207 @@ Time filter
+<td valign="top"><a href="#timefilter">TimeFilter</a>!</td>
+<td>
+
+Time filter
```
Some descriptions have one or more sentences with punctuation while some descriptions like this have just two words and no punctuation.

To be consistent, convert sentences like this to sentences by adding an article in front and punctuation.

e.g., this description can be changed to "The time filter.".

This comment applies to all the similar descriptions in this document.

RAVEENSR commented:
@RukshiW Addressed the review comments via [eebb1ee](https://github.com/wso2/docs-choreo-dev/pull/463/commits/eebb1ee6adc4e0774ffef6b2491305ab583d594c)

RukshiW commented on en/docs/insights/choreo-insights-api.md:
```diff
@@ -4221,15 +4566,193 @@ Represents an error type.
+
+### Metric
+
+Represents a metric used in CustomReports.
```
<code>CustomReports</code>?

RAVEENSR commented:
@RukshiW Fixed latest comments via [1a88d66](https://github.com/wso2/docs-choreo-dev/pull/463/commits/1a88d6634c7bc1c3db5ad7e5eae87eb6db5065ac)
